scheduler/src/cook/mesos/schema.clj

```diff
@@ -427,20 +441,20 @@
              - task succeeded => job completed
              - task failed, no other tasks, retries exceeded => job completed
              - task failed, no other tasks, retries remaining => job waiting
-             - task failed, other tasks running => job running"
+             - task failed, other tasks running => job running
+
+             Note that backfilled running instances are treated as if they don't exist."
```

Comment:
A backfilled running instance will not allow the job to be moved from waiting -> running correct? 

So one of my concerns above is not a problem after all. However, now I'm concerned that this is quite complicated. Let me think about how to simplify it.